### ISSUE TITLE
Error when using Wrangler or Cloudflare workers

### ISSUE THREAD

**Issue by kristian-puccio** [brillout/vite-plugin-ssr#911]
### Description

/src/pages/category/index.page.server.tsx?extractAssets&lang.tsx
    at createErrorWithCleanStackTrace (core:user:kb-worker:672:17)
    at assert (core:user:kb-worker:880:99)
    at getManifestEntry (core:user:kb-worker:24864:30)
    at core:user:kb-worker:24967:75
    at Array.forEach (<anonymous>)
    at retrieveAssetsProd (core:user:kb-worker:24955:28)
    at getPageAssets (core:user:kb-worker:25117:67)
    at Object.__getPageAssets (core:user:kb-worker:25529:74)
    at getHtmlTags (core:user:kb-worker:22825:46)
    at injectHtmlTagsToString (core:user:kb-worker:22962:62)
    at renderDocumentHtml (core:user:kb-worker:23786:80)
    at async executeOnRenderHtmlHook (core:user:kb-worker:25855:28)
    at async renderPageContext (core:user:kb-worker:25953:34)
    at async renderPageAttempt (core:user:kb-worker:27082:40)
    at async renderPage2 (core:user:kb-worker:26974:35)
    at async handleSsr (core:user:kb-worker:27396:25)
    at async handleFetchEvent (core:user:kb-worker:27478:25)
    at async jsonError (core:user:kb-worker:276:20)

Works perfectly using Express in dev mode.

Might be due to custom route defined in `index.page.route.ts` 
`export default '/@langId/category/@categoryId';`

**Comment by kristian-puccio** [comment]
Also I get a second error when the _error.page try's to render

Error: [vite-plugin-ssr@0.4.127][Wrong Usage] No client-side `render()` hook found. See https://vite-plugin-ssr.com/render-modes for more information. Loaded client-side page files (none of them `export { render }`):
 (1): /renderer/_error.page.tsx
    at createErrorWithCleanStackTrace (core:user:kb-worker:672:17)
    at assertUsage (core:user:kb-worker:897:96)
    at assertClientSideRenderHook (core:user:kb-worker:24558:35)
    at isHtmlOnly (core:user:kb-worker:24536:15)
    at analyzeExports (core:user:kb-worker:24531:30)
    at analyzePageClientSide (core:user:kb-worker:24626:85)
    at analyzePage (core:user:kb-worker:25481:68)
    at loadPageFilesServer (core:user:kb-worker:25511:156)
    at async renderPageContext (core:user:kb-worker:25931:27)
    at async renderPage2 (core:user:kb-worker:27024:36)
    at async handleSsr (core:user:kb-worker:27396:25)
    at async handleFetchEvent (core:user:kb-worker:27478:25)
    at async jsonError (core:user:kb-worker:276:20)

I am trying to use SSR only mode, so maybe I have set something up wrong?

**Comment by kristian-puccio** [comment]
Actually I fixed the error by turning off this option in vite.config.ts
`includeAssetsImportedByServer: true`

**Comment by brillout** [comment]
> I am trying to use SSR only mode, so maybe I have set something up wrong?

Make sure to not have any `.page.js` file (you should only have `.page.server.js` and optionally `.page.client.js` files for injecting vanilla JavaScript code).

The error message isn't really helpful in your situation. That's something the V1 design will improve on.

As for the first erorr, I'll do some digging later today.

**Comment by brillout** [comment]
Can you generate the error again with the latest version and report back the new error message? It will include helpful debug information that will help me debug this.

**Comment by brillout** [comment]
> Can you generate the error again with the latest version and report back the new error message? It will include helpful debug information that will help me debug this.

The first error message.

**Comment by kristian-puccio** [comment]
Amazing thanks for looking into this, here's the error with the new version

```
`{"manifestKeys":["_chunk-11627772.js","_chunk-11a1b4ef.js","default","node_modules/.pnpm/vite-plugin-ssr@0.4.129_vite@4.3.8/node_modules/vite-plugin-ssr/dist/esm/client/entry.js","node_modules/.pnpm/vite-plugin-ssr@0.4.129_vite@4.3.8/node_modules/vite-plugin-ssr/dist/esm/client/router/entry.js","renderer/_default.page.server.tsx?extractAssets&lang.css","renderer/_default.page.server.tsx?extractAssets&lang.tsx","renderer/_error.page.tsx","src/pages/home/index.page.server.tsx?extractAssets&lang.css","src/pages/home/index.page.server.tsx?extractAssets&lang.tsx"],"id":"/src/pages/category/index.page.server.tsx?extractAssets&lang.tsx"}`
    at createErrorWithCleanStackTrace (core:user:kb-worker:672:17)
    at assert (core:user:kb-worker:880:99)
    at getManifestEntry (core:user:kb-worker:24911:30)
    at core:user:kb-worker:25025:75
    at Array.forEach (<anonymous>)
    at retrieveAssetsProd (core:user:kb-worker:25013:28)
    at getPageAssets (core:user:kb-worker:25175:67)
    at Object.__getPageAssets (core:user:kb-worker:25587:74)
    at getHtmlTags (core:user:kb-worker:22874:46)
    at injectHtmlTagsToString (core:user:kb-worker:23011:62)
    at renderDocumentHtml (core:user:kb-worker:23835:80)
    at async executeOnRenderHtmlHook (core:user:kb-worker:25819:28)
    at async renderPageContext (core:user:kb-worker:26022:34)
    at async renderPageAttempt (core:user:kb-worker:27151:40)
    at async renderPage2 (core:user:kb-worker:27043:35)
    at async handleSsr (core:user:kb-worker:27466:25)
    at async handleFetchEvent (core:user:kb-worker:27557:25)
    at async jsonError (core:user:kb-worker:276:20)
```

**Comment by kristian-puccio** [comment]
Oh I didn't mention that with includeAssetsImportedByServer enabled the default page works perfectly but all other routes give the above error

**Comment by brillout** [comment]
> ```
> {
>   "manifestKeys": [
>     "_chunk-11627772.js",
>     "_chunk-11a1b4ef.js",
>     "default",
>     "node_modules/.pnpm/vite-plugin-ssr@0.4.129_vite@4.3.8/node_modules/vite-plugin-ssr/dist/esm/client/entry.js",
>     "node_modules/.pnpm/vite-plugin-ssr@0.4.129_vite@4.3.8/node_modules/vite-plugin-ssr/dist/esm/client/router/entry.js",
>     "renderer/_default.page.server.tsx?extractAssets&lang.css",
>     "renderer/_default.page.server.tsx?extractAssets&lang.tsx",
>     "renderer/_error.page.tsx",
>     "src/pages/home/index.page.server.tsx?extractAssets&lang.css",
>     "src/pages/home/index.page.server.tsx?extractAssets&lang.tsx"
>   ],
>   "id": "/src/pages/category/index.page.server.tsx?extractAssets&lang.tsx"
> }
> ```

That's quite unexpected: this means that the category page is missing in your `dist/client/manifest.json`. Do you have a guess why that might be?

I think the best at this point is a reproduction. It doesn't have to be minimal. If you want to create one from scratch: it doesn't have to include wrangler, the fact that the page is missing in `manifest.json` is enough.

**Comment by kristian-puccio** [comment]
You are correct it's missing from client/manifest.json, I'll dig in a bit more in a bit and try to get you a reproduction

**Comment by brillout** [comment]
👌

**Comment by kristian-puccio** [comment]
Ok got a repo, sorry about the delay.

https://github.com/kristian-puccio/vite-plugin-ssr-worker-test

I've included the `dist` folder for you.
In this case the category page is there and working fine but home page is missing.

**Comment by brillout** [comment]
Fix released in `0.4.131`.

Btw. in case you can convince your company to sponsor: https://github.com/sponsors/brillout.

**Comment by kristian-puccio** [comment]
Legend that works!
  
​
  
​I'll ask work for sure   👍
  
​

**Comment by brillout** [comment]
> ​​I'll ask work for sure 👍​

🙏 Keep me udpated 👀, also if it's a no.
